packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -231,6 +231,74 @@ class _ProductListPageState extends State<ProductListPage>
         actionModeTitle: Text(appLocalizations.compare_products_appbar_title),
         actionModeSubTitle:
             Text(appLocalizations.compare_products_appbar_subtitle),
+        actionModeActions: <Widget>[
+          IconButton(
+            icon: const Icon(Icons.delete),
+            onPressed: () async {
+              if (_selectedBarcodes.isNotEmpty) {
+                await showDialog<void>(
+                  context: context,
+                  builder: (BuildContext context) {
+                    return SmoothAlertDialog(
+                      body: Container(
+                        padding: const EdgeInsets.only(left: SMALL_SPACE),
+                        child: Column(
+                          crossAxisAlignment: CrossAxisAlignment.start,
+                          children: <Widget>[
+                            Text(
+                              appLocalizations.alert_clear_selected_user_list,
+                            ),
+                            const SizedBox(
+                              height: SMALL_SPACE,
+                            ),
+                            Text(
+                              appLocalizations.confirm_clear_selected_user_list,
+                            ),
+                          ],
+                        ),
+                      ),
+                      positiveAction: SmoothActionButton(
+                        onPressed: () async {
+                          await daoProductList.bulkSet(
+                            productList,
+                            _selectedBarcodes.toList(growable: false),
+                            include: false,
+                          );
+                          await daoProductList.get(productList);
+                          if (!mounted) {
+                            return;
+                          }
+                          setState(() {});
```

Comment:
Calling setState and just after closing the screen. I don't think this one is useful.
Was-it already in the code?

Reply:
@g123k if we don't call setState() then the product will be deleted but the list will still show the deleted product and the user will have to press a back button to referesh the state.

Reply:
True, ugly code, but OK